### PR TITLE
JSON Schema: Cleanup and documentation

### PR DESCRIPTION
**Pull request by smyrman**
Cleaned up the JSON Schema package documentation a bit, as well as some
of the code. I changed one of the tests and the ErrNotImplemented error
message. Pacakge impors was rearanged by goimports.

I used the Go standard library as a guide line for the docs, including
the use of full sentences with a trailing period in comments and short,
lower case error messages.

### CODE REVIEW

File: schema/encoding/jsonschema/jsonschema.go [hunk 1]
```diff
@@ -166,29 +168,32 @@ func schemaToJSONSchema(w io.Writer, s *schema.Schema) (err error) {
...
-	ew.writeFormat(`, "required": [%s]`, strings.Join(required, ", "))
+
+	if len(required) > 0 {
+		ew.writeFormat(`, "required": [%s]`, strings.Join(required, ", "))
+	}
 	return ew.err
 }
 
-// Encoder encodes schema.Schema into a JSONSchema
+// Encoder writes the JSON Schema representation of a schema.Schema to an output stream. Note that only a sub-set of the
```
Comment (smyrman) [inline schema/encoding/jsonschema/jsonschema.go]: Broke the lines at 120 chars. @rs, is there any particular line width you prefer or use elsewhere in this project?

**Comment by yanfali** [comment]
That looks pretty good. Is the required array optional if empty?

File: schema/encoding/jsonschema/jsonschema_test.go [hunk 2]
```diff
@@ -413,42 +414,38 @@ func TestArrayOfObjects(t *testing.T) {
 			},
 		},
 	}
+	const expect = `{
```
Comment (yanfali) [inline schema/encoding/jsonschema/jsonschema_test.go]: much cleaner

**Comment by smyrman** [comment]
> That looks pretty good. Is the required array optional if empty?
I thought so, but let me double check it...

**Comment by smyrman** [comment]
>> That looks pretty good. Is the required array optional if empty?

> I thought so, but let me double check it...

Yes, looks like It is optional.

According to the [Core/Validation Meta-Schema](http://json-schema.org/schema) (which is the JSON Schema for JSON Schema), there are no required fields for a "JSON Schema" -- even "type" is optional.

**Comment by yanfali** [comment]
LGTM. Do you want to add a test for required not being in output?

**Comment by smyrman** [comment]
Sure, let's add it here.

Other than that, I hope to add more tests through #44 :-)

**Comment by smyrman** [comment]
> Sure, let's add it here.

Actually, we already have it! Notice that the outer schema  in `TestArrayOfObjects()` does not have the required array...
```
const expect = `{
	"type": "object",
	"title": "A list of students",
	"properties": {
		"students": {...}
	}
	// <-- no required list
}`
```

I can add it as a separate test in #44.

**Comment by yanfali** [comment]
Sounds good

**Comment by smyrman** [comment]
@rs, this one is ready if you are happy with it.

File: schema/encoding/jsonschema/jsonschema.go (hunk 1 above)
Comment (rs) [inline schema/encoding/jsonschema/jsonschema.go]: I don't particularly care, 120 is fine.

**Pull request merged**